GameData.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GameOfLife
+{
+    /// <summary>
+    /// Data of application.
+    /// </summary>
+    public class GameData
+    {
+        readonly public int[,] data;
+        readonly public int[,] newData;
+
+        /// <summary>
+        /// Data of application.
+        /// </summary>
+        public GameData()
+        {
+            int column = UserInterface.GetValueInRange("Input amount of columns", 5, 200);
+            int row = UserInterface.GetValueInRange("Input amount of rows", 5, 200);
+            data = new int[row, column];
+            newData = new int[row, column];
+            FillDefault();
+        }
+
+        /// <summary>
+        /// Fills gameboard with random alive cells.
+        /// </summary>
+        void FillDefault()
+        {
+            Random random = new Random();
+            for (int i = 0; i < data.GetLength(0); i++)
+            {
+                for (int j = 0; j < data.GetLength(1); j++)
+                {
+                    data[i, j] = random.Next(0, 4) == 0 ? 1 : 0;
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Iterates to the next generation.
+        /// </summary>
+        public void Iterate()
+        {
+            for (int row = 0; row < data.GetLength(0); row++)
+            {
+                for (int column = 0; column < data.GetLength(1); column++)
+                {
+                    newData[row, column] = IsLiveCell(row, column) ? 1 : 0;
+                }
+            }
+            TransferData();
```

Comment:
why do you do the same in TransferData method as in this but the only change is in which array the data is saved, you know that you can write one method that iterates through your array and updates the values and then just pass the needed array.

Reply:
I use two arrays so to not make changes on cells in a line, that could be neighbors to a cell that has not been updated yet. I can add the other for cycle in the same function if that would be better?

Reply:
from what I see all you do is update one array, when all the updates are done, then you taka that array and write the values to the needed one, the problem is that you could update the values already in the first one when you run your loops, because now instead of looping once for the array value change you loop two times, which is not a good approach as that potentially could impact performance.

Reply:
I removed the TransferData method, made the 2nd array in the GetNewGeneration method and added array.Copy as a substitute.